### PR TITLE
Odst 253 resolution tags for responses

### PR DESCRIPTION
**Pull request by BrendenMcGrew**
# To be Completed by the Pull Requester

### Checklist:

- [x] I have performed a self-review of my code to ensure that unintended code is not introduced.
- [x] I have commented my code, particularly in hard-to-understand areas to ensure that the next group of developers can understand my code.
- [x] I have added tests (if appropriate) that prove my fix/feature functions so that future modifications to the code do not break my additions.

### Description:

Tægs

### Type of change (Pick Any)

1. Bug fix. Non-breaking change which fixes an issue, example: swapping the order of numbers on a UI.

2. New feature. Non-breaking change which adds functionality, example: new page on the frontend.

### How Has This Been Tested?

Describe the tests that you created if applicable. Use N/A if no tests were created.

-Tests for the new tag component

-Back end tests for the tags

-Edited end to end tests to include tags

---

# To be Completed By Reviewers

Reviewer # 01
- [x] I have performed a review of the code and checked for possible unintentional edits.
- [x] I concur with the amount of comments made.
- [x] I concur with the amount of tests made.

Reviewer # 02
- [x] I have performed a review of the code and checked for possible unintentional edits.
- [x] I concur with the amount of comments made.
- [x] I concur with the amount of tests made.

### CODE REVIEW

**Review by ericcater — APPROVED** [review]

**Review by yurisim — CHANGES_REQUESTED** [review]
Fix conflicts

**Review by ericcater — APPROVED** [review]

File: apps/ods/src/app/responses/responses.component.ts [hunk 1]
```diff
@@ -61,13 +55,21 @@ export class ResponsesComponent implements OnInit {
...
-  @ViewChild('tagInput') tagInput: ElementRef<HTMLInputElement>;
+  constructor(
+    private fb: UntypedFormBuilder,
+    private responsesService: ResponsesService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {}
 
-  async ngOnInit() {
+  //region Main functions
```
Comment (yurisim) [inline apps/ods/src/app/responses/responses.component.ts]: Either remove this or add additional regions but this doesn't block the PR, just something I noticed.

File: apps/ods/src/app/responses/select-tags/select-tags.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    //Triggers the input event so generateFilteredTags is ran and detects the value
+    component.tagInput.nativeElement.dispatchEvent(new Event('input'));
+    component.generateFilteredTags();
+
+    const actual: string[] = component.filteredTags;
+    const expected: string[] = ['Tag 5'];
+
+    console.log(actual.length);
+    console.log(expected.length);
```
Comment (yurisim) [inline apps/ods/src/app/responses/select-tags/select-tags.component.spec.ts]: Remove these console.logs

File: apps/ods/src/app/responses/select-tags/select-tags.component.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  ElementRef,
+  Output,
+  EventEmitter,
+} from '@angular/core';
+import { MatChipInputEvent } from '@angular/material/chips';
+import { MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { FormControl } from '@angular/forms';
+
+@Component({
+  // eslint-disable-next-line @angular-eslint/component-selector
```
Comment (yurisim) [inline apps/ods/src/app/responses/select-tags/select-tags.component.ts:14]: what is this disabling?

File: apps/ods/src/app/responses/select-tags/select-tags.component.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  tagCtrl = new FormControl();
+
+  filteredTags;
+
+  @ViewChild('tagInput') tagInput: ElementRef<HTMLInputElement>;
+
+  /**
+   * User selects a tag to be added, which must be a part of the possible tags and not the selected tags
+   * The input box will be cleared after the event has been passed back to the parent component
+   * @param event
```
Comment (yurisim) [inline apps/ods/src/app/responses/select-tags/select-tags.component.ts]: these params can still be commented. Some of thee events can only be triggered from certain controls but we merge events at some point.

File: src/apps/ods-api/src/tag/tag.module.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { Module } from '@nestjs/common';
+
+@Module({})
+export class TagModule {}
```
Comment (yurisim) [inline src/apps/ods-api/src/tag/tag.module.ts]: You def did not mean to commit this

**Review by yurisim — CHANGES_REQUESTED** [review]
There are some unintended changes in this.

**Review by Mikehickey7 — APPROVED** [review]
looks amazing, i found no errors, but fix the errors sim found

File: apps/ods/src/app/responses/select-tags/select-tags.component.ts (hunk 3 above)
Comment (BrendenMcGrew) [reply apps/ods/src/app/responses/select-tags/select-tags.component.ts:14]: The linter wants custom selectors to have a name starting with the name of the project

File: src/apps/ods-api/src/tag/tag.module.ts (hunk 5 above)
Comment (BrendenMcGrew) [reply src/apps/ods-api/src/tag/tag.module.ts]: You are correct, sir

**Review by Mikehickey7 — APPROVED** [review]

File: apps/ods/src/app/responses/select-tags/select-tags.component.ts (hunk 4 above)
Comment (BrendenMcGrew) [reply apps/ods/src/app/responses/select-tags/select-tags.component.ts]: I added two param comments and described what both did

File: apps/ods/src/app/responses/select-tags/select-tags.component.spec.ts (hunk 2 above)
Comment (BrendenMcGrew) [reply apps/ods/src/app/responses/select-tags/select-tags.component.spec.ts]: Done

File: apps/ods/src/app/responses/responses.component.ts (hunk 1 above)
Comment (BrendenMcGrew) [reply apps/ods/src/app/responses/responses.component.ts]: I added the regions back with hashtags

**Review by yurisim — APPROVED** [review]

**Pull request merged**